### ISSUE TITLE
Rspack plugin support

### ISSUE THREAD

**Issue by blueagler** [facebook/stylex#172]
Can we have an RSPack plugin?

**Comment by nmn** [comment]
Happy to accept contributions for this.

**Comment by necolas** [comment]
It's also fine for plugins to be developed by 3rd parties in other repos. We can't maintain plugins for every build system out there

**Comment by iowxy** [comment]
Maybe we can use [unplugin](https://github.com/unjs/unplugin) to create transpiler

What's unplugin?

Unified plugin system for build tools.

Currently supports:

[Vite](https://vitejs.dev/)
[Rollup](https://rollupjs.org/)
[Webpack](https://webpack.js.org/)
[esbuild](https://esbuild.github.io/)
[Rspack](https://www.rspack.dev/) (⚠️ experimental)

**Comment by iowxy** [comment]
> Happy to accept contributions for this. It might be possible to use `unplugin` to do this.

This is what I want to say

**Comment by eryue0220** [comment]
Can I have a try for this?

**Comment by eryue0220** [comment]
I have written the plugin through `unplugin` to support: `rspack`, `esbuild` and `vite`. The repo is [here](https://github.com/eryue0220/unplugin-stylex).

**But** there is in a primitive stage, and there are still many things to complete. And I think I will need to extra days to solve the [unit test](https://github.com/eryue0220/unplugin-stylex/issues/2) and the [ci](https://github.com/eryue0220/unplugin-stylex/issues/2) at least.

Any questions please DM me, or commit an issue in the repo.  

Thanks~

**Comment by GiancarlosIO** [comment]
Hi there!
I would like to help on this. Is there a guide on how to use unplugin to create the plugin?

**Comment by nmn** [comment]
@GiancarlosIO 

https://unplugin.unjs.io

See the Rollup plugin and the community Vite plugins for further help.

**Comment by RavenColEvol** [comment]
@nmn will it be okay to use typescript instead of flow for new package contribution? as flow has a learning curve and flow types for new packages like unplugin or rspack is not available.

**Comment by nmn** [comment]
@RavenColEvol I'm not against using TS for bundler plugins. You can also contribute TS code and I can do the work of migrating it to Flow if it turns out that having a single type system for the whole repo is helpful.

**Comment by RavenColEvol** [comment]
@nmn I later found out that unplugin community implementation exists and it's mentioned in the docs, so idk if it makes sense to create a unplugin stylex plugin in the stylex repo.

**Comment by RavenColEvol** [comment]
https://stylexjs.com/docs/learn/ecosystem/#unplugin

**Comment by nmn** [comment]
Yes, it makes most sense to contribute to that existing plugin and make it better or start another plugin yourself. Over time, we might make some community plugins official, but we don't want to take on additional maintenance burden at the moment and get out of the way of the community.
